src/config/schema.rs

```diff
@@ -119,9 +130,24 @@ pub struct Local {
     pub data_dir: String,
 }
 
+impl Local {
+    pub fn to_config(&self) -> HashMap<String, String> {
+        let mut config = HashMap::new();
+        config.insert("data_dir".to_string(), self.data_dir.clone());
+        config
+    }
+}
+
 #[derive(Deserialize, Debug, PartialEq, Eq, Clone)]
 pub struct InMemory {}
 
+impl InMemory {
+    pub fn to_config(&self) -> HashMap<String, String> {
+        // InMemory does not have any specific configurations.
+        HashMap::new()
+    }
+}
+
 #[derive(Deserialize, Debug, PartialEq, Eq, Clone)]
 pub struct S3 {
```

Comment:
@gruuya This change will be a snow ball as these structs are used in several part of the code. Would be ok to address this change in a separate PR, so that we reduce the complexity of it?

Reply:
Yup, sounds good.